stargate/pkg/auth/client.go

```diff
@@ -110,7 +126,42 @@ func (t tableBasedTokenProvider) getToken(ctx context.Context) (string, error) {
 		return "", fmt.Errorf("error unmarshalling response body: %v", err)
 	}
 
-	return ar.AuthToken, nil
+	// If the server asks, skip the cache.
+	if !t.cacheEnabled {
+		return ar.AuthToken, nil
+	}
+
+	// cache expiration default of 10min (arbitrary).
+	ccExpSec := 600
```

Comment:
Why not don't cache it if there's no `Cache-Control`? (It should be there though)